### ISSUE TITLE
Now seems like a good time for the next minor release

### ISSUE THREAD

**Issue by telamonian** [jupyterlab/jupyterlab-git#453]
A whole raft of things have been changed, fixed, or improved since the last release. We should go ahead and do a new minor release.

The recent issue #452 seems like a good indication that we should go ahead and publish a new version already. There's a few cool new features on the near horizon (plain-text diff, simple staging, better branch handling), but there's no reason to let them block a new minor release in the meantime.

@fcollonval Given the CD stuff you've been working on, do you want to handle the release? Otherwise, I can just do it manually

**Comment by fcollonval** [comment]
Hey @telamonian I fully agree and I gladly do it. But I don't have the rights on pypi nor on npmjs to publish this package.

**Comment by telamonian** [comment]
Then I'll take care of it. I have rights on npmjs, and I'll see about pypi. My plan will be that both server extension/labextension will get bumped to `0.9.0`.

It's optional, but it would be nice to have a changelog. I can write up a summary of my own PRs. @fcollonval @jaipreet-s and anyone else who's done a recent PR: can you please post a short summary of any PRs you posted after `0.8.2` was released (on 09/05/2019)?

**Comment by jaipreet-s** [comment]
@telamonian Thanks for owning the release. Whats your PyPI username so I can grant you permissions to upload?

**Comment by telamonian** [comment]
You're welcome. The recent progress has been good, and I think we're well on the path to `v1.0`. Still needs a bunch of fixes and a ton of elbow grease, but we're getting there.

On pypi, my user name is just tel:

https://pypi.org/user/tel/

**Comment by jaipreet-s** [comment]
Done.

**Comment by telamonian** [comment]
I've been working on this for the last couple of days. Before we do the release, there's one PR that needs to go in (in one form or another): #458.

Right now, we include in our Python package (ie the sdist) the complete build system for the labextension, *as well as* a built labextension artifact (ie the `.tar.gz` that `npm pack` spits out). This is completely redundant, and leads to some very questionable behavior (in order to make a wheel from the sdist, a totally pointless labextension rebuild is carried out). Among other issues, this means that some subset of user will require `node` in order to install our python package.

#458 rearranges things so that just the built/packaed labextension `.tar.gz` is included in the sdist, and no other related bits.

**Comment by telamonian** [comment]
As per ongoing discussions with @blink1073 and @saulshanabrook, big changes are coming soon the jupyterlab extension system as a whole. The future shape of extensions will be similar to that of #458: a Python package that also includes a prebuilt, ready to use JS bundle.

More details on that will come after the dev meeting next week

**Comment by kgryte** [comment]
@telamonian Would it be possible to sneak in #459,  provided it is deemed acceptable, before publishing a new release?

**Comment by telamonian** [comment]
The new build is now live on the Pypi test server (but not the real Pypi or npmjs) if anyone wants to give it a whirl:

https://test.pypi.org/project/jupyterlab-git/

I've tested it and it seems good, aside from the bug in simple staging that I just found: #460

**Comment by telamonian** [comment]
@kgryte Since it's short, I already reviewed #459. If it's done when the release is made, it'll go in (no promises)

**Comment by telamonian** [comment]
Okay, so with the changes made in #458 I think our unified serverextension/labextension bundle (and the build/release system that goes with it) is up to snuff. My current plan is to do the release of `0.9.0` in the next day or two.

I'll also try to get #459 and #469 in because they're almost done, but I won't let them block the release.

**Comment by telamonian** [comment]
I just put the `0.9.0rc0` up on the test Pypi. Try it out with:

```
pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple jupyterlab-git==0.9.0rc0
```

Assuming there's no glaring issues, I'll release `0.9.0` on the real Pypi in the next hour

**Comment by telamonian** [comment]
Me and @kgryte tested out the RC, and we did indeed find a couple of glaring issues. The fixes are now up at #477 and #478. Once these get pulled in I'll do the release

**Comment by telamonian** [comment]
`v0.9.0` is live! May the good OSS mojo be with us
